core/src/main/java/org/apache/calcite/rel/metadata/RelMetadataQueryBase.java

```diff
@@ -75,11 +78,20 @@
       new ThreadLocal<>();
 
   //~ Constructors -----------------------------------------------------------
-
+  @Deprecated // to be removed before 2.0
   protected RelMetadataQueryBase(@Nullable JaninoRelMetadataProvider metadataProvider) {
```

Comment:
It is a public interface (public class, protected signature). While I agree that it would be unusual for a downstream project to call this, I don't think we should break public interfaces unless it is absolutely necessary.